src/inmanta/references.py

```diff
@@ -313,15 +339,17 @@ def is_dataclass_reference(cls) -> bool:
         return dataclasses.is_dataclass(cls.get_reference_type())
 
     @abc.abstractmethod
-    def resolve(self) -> T:
+    def resolve(self, logger: "handler.LoggerABC") -> T:
         """This method resolves the reference and returns the object that it refers to"""
+        pass
 
-    def get(self) -> T:
+    def get(self, logger: "handler.LoggerABC") -> T:
         """Get the value. If we have already resolved it a cached value is returned, otherwise resolve() is called"""
         if not self._reference_value_cached:
-            self._reference_value = self.resolve()
+            self._reference_value = self.resolve(logger)
             self._reference_value_cached = True
 
+        logger.debug("Using cached value for reference %(reference)s", reference=str(self))
```

Comment:
```suggestion
        else:
            logger.debug("Using cached value for reference %(reference)s", reference=str(self))
```

Reply:
logger and the cache are both per-resource